civis/io/_tables.py

```diff
@@ -782,18 +830,76 @@ def _get_sql_select(table, columns=None):
     return sql
 
 
-def _download_file(url, local_path):
-    response = requests.get(url, stream=True)
-    response.raise_for_status()
+def _get_headers(client, sql, database, credential_id):
+    sql = 'select * from ({}) limit 1'.format(sql)
```

Comment:
If we have a complex query, it is possible it will be slower to run the query twice than unload with `PARALLEL OFF` ?

Reply:
limit 1 should make it very fast

Reply:
If there's an order by, it could still be very slow, no?

Reply:
agreed. i'll check for order bys and we'll do a regular export. platform will not do a parallel unload for these anyway so there is no reason for the client to retrieve headers separately

Reply:
- added 'begin read only;' to _get_headers function
- added separate function _include_header to determine if we should request headers